### ISSUE TITLE
Deterministic training guidelines for different common workflows. 

### ISSUE THREAD

**Issue by yshvrdhn** [pytorch/vision#7457]

**Comment by yshvrdhn** [comment]
@datumbox I agree I have ensured in training that dataloader give same samples and verified it. However even after using ```
torch.use_deterministic_algorithms() ``` it does not flag pytorch operations that are not deterministic as mentioned in the [reproducibility  docs ](https://pytorch.org/docs/stable/notes/randomness.html). Its mentioned [here](https://github.com/facebookresearch/detectron2/issues/4723) that atomic add in roi align is most likely the issue. We should at least mention these in the docs for popular vision models.
 It would be great if more each vision model we have a placeholder of determinism its a critical component in using pytorch in production or in research.

**Comment by NicolasHug** [comment]
@yshvrdhn would you have a small reproducible example where we can reproduce a significant output difference due to un-controllable randomness?

The randomness from the atomic operator is generally due to the fact that the order of the operations matter (i.e. `a + b != b + a`). But it sounds surprising that such small arithmetic differences in `roi_align` would lead to significant output differences, unless the model is not well behaved.


> We should at least mention these in the docs for popular vision models so that we can make an informed opinion if the said model training is deterministic or not and if they will be fixed in the future following the guides by pytorch.

If the randomness only comes from atomic operators, only the RCNN family should be affected (Faster-RCNN, Mask-RCNN and KeyPoint-RCNN).
